Sources/Sentry/Public/SentryDefines.h

```diff
@@ -1,5 +1,11 @@
 #import <Foundation/Foundation.h>
 
+// SentryDefines.h is a key header and will be checked early,
+// ensuring this error appears first during the compile process.
```

Comment:
`m`: Please explain the detailed background why this is required including the weird side effect of 
 APPLICATION_EXTENSION_API_ONLY_NO and maybe even some links to the original issues, cause this can be quite confusing.

Reply:
Added in https://github.com/getsentry/sentry-cocoa/pull/4603/commits/5e14a9d18a78cbca8527d08ae237a005a4434c99

Reply:
Maybe also briefly mention the problems it caused on RN with a link to the issue.